File: 2483-minimum-penalty-for-a-shop/2483-minimum-penalty-for-a-shop.cpp
class Solution {
public:
    int bestClosingTime(string customers) {
        int n = customers.size();
    
        int penalty=0;
        int minP=penalty, minI=0;
        for (int i=1; i <= n; i++) {
            int y=customers[i-1] == 'Y'?1:-1;
            penalty-=y;
            if (minP> penalty) {
                minP=penalty;
                minI=i;
            }
        }
        return minI;
    }
};

File: 2483-minimum-penalty-for-a-shop/2483-minimum-penalty-for-a-shop.ts
function bestClosingTime(customers: string): number {
     let n = customers.length;
    
        let penalty=0;
        let minP=penalty, minI=0;
        for (let i=1; i <= n; i++) {
            let y=customers[i-1] == 'Y'?1:-1;
            penalty-=y;
            if (minP> penalty) {
                minP=penalty;
                minI=i;
            }
        }
        return minI;
    
};

